# PROYECTO DE CONSOLIDACIÓN 1

Tratamiento de la información relativa al análisis

In [1]:
import pandas as pd
import numpy  as np

In [2]:
from fancyimpute import KNN
from fancyimpute import SimpleFill

Los dos primeros paquetes que hemos importado son numpy y pandas, para la manipulación general de datos.

Una vez que tenemos todas las librerías, vamos a cargar los datos en memoria.

## 2. Combinar los datos

In [14]:
df_red = pd.read_csv('C:/Users/acarreton/Desktop/Data engineering/Trabajo de consolidacion 1/PC1/Data/wine+quality/winequality-red.csv', sep=';')

In [13]:
df_white = pd.read_csv('C:/Users/acarreton/Desktop/Data engineering/Trabajo de consolidacion 1/PC1/Data/wine+quality/winequality-white.csv', sep=';')

In [15]:
df_red.head()

fixed acidity  volatile acidity  citric acid  residual sugar  chlorides  \
0            7.4              0.70         0.00             1.9      0.076   
1            7.8              0.88         0.00             2.6      0.098   
2            7.8              0.76         0.04             2.3      0.092   
3           11.2              0.28         0.56             1.9      0.075   
4            7.4              0.70         0.00             1.9      0.076   

   free sulfur dioxide  total sulfur dioxide  density    pH  sulphates  \
0                 11.0                  34.0   0.9978  3.51       0.56   
1                 25.0                  67.0   0.9968  3.20       0.68   
2                 15.0                  54.0   0.9970  3.26       0.65   
3                 17.0                  60.0   0.9980  3.16       0.58   
4                 11.0                  34.0   0.9978  3.51       0.56   

   alcohol  quality  
0      9.4        5  
1      9.8        5  
2      9.8        5  
3      9.8        6  
4      9.4        5

In [16]:
df_white.head()

fixed acidity  volatile acidity  citric acid  residual sugar  chlorides  \
0            7.0              0.27         0.36            20.7      0.045   
1            6.3              0.30         0.34             1.6      0.049   
2            8.1              0.28         0.40             6.9      0.050   
3            7.2              0.23         0.32             8.5      0.058   
4            7.2              0.23         0.32             8.5      0.058   

   free sulfur dioxide  total sulfur dioxide  density    pH  sulphates  \
0                 45.0                 170.0   1.0010  3.00       0.45   
1                 14.0                 132.0   0.9940  3.30       0.49   
2                 30.0                  97.0   0.9951  3.26       0.44   
3                 47.0                 186.0   0.9956  3.19       0.40   
4                 47.0                 186.0   0.9956  3.19       0.40   

   alcohol  quality  
0      8.8        6  
1      9.5        6  
2     10.1        6  
3      9.9        6  
4      9.9        6

Combinamos en nuevo dataframe con toda la información y añadiendo una nueva columna que distinga por tipo de vino (red o white)

In [17]:
df_red['wine_type'] = 'red'
df_white['wine_type'] = 'white'

Creamos un único dataframe con toda la información a analizar

In [18]:
total_df = pd.concat([df_red, df_white], ignore_index=True)

In [19]:
total_df.head()

fixed acidity  volatile acidity  citric acid  residual sugar  chlorides  \
0            7.4              0.70         0.00             1.9      0.076   
1            7.8              0.88         0.00             2.6      0.098   
2            7.8              0.76         0.04             2.3      0.092   
3           11.2              0.28         0.56             1.9      0.075   
4            7.4              0.70         0.00             1.9      0.076   

   free sulfur dioxide  total sulfur dioxide  density    pH  sulphates  \
0                 11.0                  34.0   0.9978  3.51       0.56   
1                 25.0                  67.0   0.9968  3.20       0.68   
2                 15.0                  54.0   0.9970  3.26       0.65   
3                 17.0                  60.0   0.9980  3.16       0.58   
4                 11.0                  34.0   0.9978  3.51       0.56   

   alcohol  quality wine_type  
0      9.4        5       red  
1      9.8        5       red  
2      9.8        5       red  
3      9.8        6       red  
4      9.4        5       red

Estudiemos las dimensiones del conjunto de datos a analizar

In [20]:
total_df.size

84461

In [22]:
total_df.shape

(6497, 13)

In [24]:
total_df.columns

Index(['fixed acidity', 'volatile acidity', 'citric acid', 'residual sugar',
       'chlorides', 'free sulfur dioxide', 'total sulfur dioxide', 'density',
       'pH', 'sulphates', 'alcohol', 'quality', 'wine_type'],
      dtype='object')

In [26]:
total_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 6497 entries, 0 to 6496
Data columns (total 13 columns):
 #   Column                Non-Null Count  Dtype  
---  ------                --------------  -----  
 0   fixed acidity         6497 non-null   float64
 1   volatile acidity      6497 non-null   float64
 2   citric acid           6497 non-null   float64
 3   residual sugar        6497 non-null   float64
 4   chlorides             6497 non-null   float64
 5   free sulfur dioxide   6497 non-null   float64
 6   total sulfur dioxide  6497 non-null   float64
 7   density               6497 non-null   float64
 8   pH                    6497 non-null   float64
 9   sulphates             6497 non-null   float64
 10  alcohol               6497 non-null   float64
 11  quality               6497 non-null   int64  
 12  wine_type             6497 non-null   object 
dtypes: float64(11), int64(1), object(1)
memory usage: 660.0+ KB


Tenemos un total de 84.461 datos. Se puede entender como 13 variables y 6497 elementos en la muestra de los que se obtiene, para cada uno de los 6497 elementos de la muestra, las 13 cualidades.
Tenemos 13 variables de las cuales 12 son cuantitativas (11 float y 1 entera) y 1 es cualitativa (el color del vino).

## 3. Filtrar atípicos y manejar datos ausentes

Análisis estadístico o inspección visual de cada columna numérica para identificar valores atípicos.

In [23]:
total_df.describe(include="all")

fixed acidity  volatile acidity  citric acid  residual sugar  \
count     6497.000000       6497.000000  6497.000000     6497.000000   
unique            NaN               NaN          NaN             NaN   
top               NaN               NaN          NaN             NaN   
freq              NaN               NaN          NaN             NaN   
mean         7.215307          0.339666     0.318633        5.443235   
std          1.296434          0.164636     0.145318        4.757804   
min          3.800000          0.080000     0.000000        0.600000   
25%          6.400000          0.230000     0.250000        1.800000   
50%          7.000000          0.290000     0.310000        3.000000   
75%          7.700000          0.400000     0.390000        8.100000   
max         15.900000          1.580000     1.660000       65.800000   

          chlorides  free sulfur dioxide  total sulfur dioxide      density  \
count   6497.000000          6497.000000           6497.000000  6497.000000   
unique          NaN                  NaN                   NaN          NaN   
top             NaN                  NaN                   NaN          NaN   
freq            NaN                  NaN                   NaN          NaN   
mean       0.056034            30.525319            115.744574     0.994697   
std        0.035034            17.749400             56.521855     0.002999   
min        0.009000             1.000000              6.000000     0.987110   
25%        0.038000            17.000000             77.000000     0.992340   
50%        0.047000            29.000000            118.000000     0.994890   
75%        0.065000            41.000000            156.000000     0.996990   
max        0.611000           289.000000            440.000000     1.038980   

                 pH    sulphates      alcohol      quality wine_type  
count   6497.000000  6497.000000  6497.000000  6497.000000      6497  
unique          NaN          NaN          NaN          NaN         2  
top             NaN          NaN          NaN          NaN     white  
freq            NaN          NaN          NaN          NaN      4898  
mean       3.218501     0.531268    10.491801     5.818378       NaN  
std        0.160787     0.148806     1.192712     0.873255       NaN  
min        2.720000     0.220000     8.000000     3.000000       NaN  
25%        3.110000     0.430000     9.500000     5.000000       NaN  
50%        3.210000     0.510000    10.300000     6.000000       NaN  
75%        3.320000     0.600000    11.300000     6.000000       NaN  
max        4.010000     2.000000    14.900000     9.000000       NaN

In [27]:
# Eliminamos los vacíos en caso de que los haya y repetimos el análisis estadístico
total_df_clean=total_df.dropna()
total_df_clean.describe(include='all')

fixed acidity  volatile acidity  citric acid  residual sugar  \
count     6497.000000       6497.000000  6497.000000     6497.000000   
unique            NaN               NaN          NaN             NaN   
top               NaN               NaN          NaN             NaN   
freq              NaN               NaN          NaN             NaN   
mean         7.215307          0.339666     0.318633        5.443235   
std          1.296434          0.164636     0.145318        4.757804   
min          3.800000          0.080000     0.000000        0.600000   
25%          6.400000          0.230000     0.250000        1.800000   
50%          7.000000          0.290000     0.310000        3.000000   
75%          7.700000          0.400000     0.390000        8.100000   
max         15.900000          1.580000     1.660000       65.800000   

          chlorides  free sulfur dioxide  total sulfur dioxide      density  \
count   6497.000000          6497.000000           6497.000000  6497.000000   
unique          NaN                  NaN                   NaN          NaN   
top             NaN                  NaN                   NaN          NaN   
freq            NaN                  NaN                   NaN          NaN   
mean       0.056034            30.525319            115.744574     0.994697   
std        0.035034            17.749400             56.521855     0.002999   
min        0.009000             1.000000              6.000000     0.987110   
25%        0.038000            17.000000             77.000000     0.992340   
50%        0.047000            29.000000            118.000000     0.994890   
75%        0.065000            41.000000            156.000000     0.996990   
max        0.611000           289.000000            440.000000     1.038980   

                 pH    sulphates      alcohol      quality wine_type  
count   6497.000000  6497.000000  6497.000000  6497.000000      6497  
unique          NaN          NaN          NaN          NaN         2  
top             NaN          NaN          NaN          NaN     white  
freq            NaN          NaN          NaN          NaN      4898  
mean       3.218501     0.531268    10.491801     5.818378       NaN  
std        0.160787     0.148806     1.192712     0.873255       NaN  
min        2.720000     0.220000     8.000000     3.000000       NaN  
25%        3.110000     0.430000     9.500000     5.000000       NaN  
50%        3.210000     0.510000    10.300000     6.000000       NaN  
75%        3.320000     0.600000    11.300000     6.000000       NaN  
max        4.010000     2.000000    14.900000     9.000000       NaN

De esto se deduce que no hay NaNs

In [ ]:
chat